troubleshooting: add s3-related entries (#1725)
iterative/dvc.org: content/docs/user-guide/troubleshooting.md

with `ulimit` on UNIX-like system (for example `ulimit -n 1024`), or
[increasing Handles limit](https://blogs.technet.microsoft.com/markrussinovich/2009/09/29/pushing-the-limits-of-windows-handles/)
on Windows. Otherwise, please try using a lower `JOBS` value.

> ## Unable to find credentials {#no-credentials}

Reviewer: ```suggestion
## Unable to find credentials {#no-aws-credentials}
```

Reviewer: Never mind about this one (per https://github.com/iterative/dvc/pull/4480#pullrequestreview-476372465).

Make sure that you have your AWS credentials setup either through
[usual AWS configuration](https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-files.html)

> or using [DVC configs](/doc/command-reference/remote/modify).

Reviewer: ```suggestion
or by setting `access_key_id` and `secret_access_key` with `dvc remote modify`.
```

Reviewer: I'll commit this just to see what happens to the next buid.

## Unable to connect {#connection-error}

> Make sure you have a working internet connection and you are able to access your

Reviewer: ```suggestion
Make sure you are online and able to access your
```

Reviewer: (and this one since a similar suggestion was already applied in the core repo PR)

> [AWS S3 endpoint](https://docs.aws.amazon.com/general/latest/gr/s3.html) or, if
> [explicitly configured](/doc/command-reference/remote/modify), your
> `endpointurl`.

Reviewer: ```suggestion
[AWS S3 endpoint](https://docs.aws.amazon.com/general/latest/gr/s3.html), or
`endpointurl` if explicitly set with `dvc remote modify`.
```

## Bucket does not exist {#no-bucket}

> Make sure
> [your bucket exists](https://docs.aws.amazon.com/AmazonS3/latest/user-guide/create-bucket.html)
> in the correct
> [`region` and/or `endpointurl`](/doc/command-reference/remote/modify).

Reviewer: ```suggestion
Make sure your bucket
[exists](https://docs.aws.amazon.com/AmazonS3/latest/user-guide/create-bucket.html)
in the correct `region` and/or `endpointurl` (see `dvc remote modify`).
```
